### PR TITLE
Social links

### PR DESCRIPTION
**Pull request by Kuriv2001**
### Social Media Links

I added the capability to have social media links, that are associated with events. 

1. They can be selected on the last page of the event creation screen (directly by username/invite code, so no random website, possibly malicious, can be specified):
Screenshot 2024-05-05 at 13 10 54

2. Depending on if a value has been set in the previous panel, the social media links that have a value are display in the viewDetailedEventScreen. When clicked, the corresponding link will be opened:
Screenshot 2024-05-05 at 13 10 42

3. They can then also be changed by editing the event.

I also made some changes in the ViewModel to support the new social media links.

Tests have also been updated to comply with changes and new tests have been added for new functionality.

### CODE REVIEW

**Review by QuicheLorraine3600 — CHANGES_REQUESTED** [review]
This is a very nice work, but after a small discussion with the team, we have concluded that putting a Instagram and a Discord username per event isn't really what we need.
We want to have urls to Whatsapp or Instagram groups per events to make sure guests and staffs can chat with each other.
The changes required are minimal, you just have to change the placeholder asking for a username to a placeholder asking for an URL.

**Comment by Kuriv2001** [comment]
Alright, I changed the fields to be able to take links now, while still making sure they're links to the correct sites and not some random ones (for security). I checked and for whatsapp there are group invite links, for instagram I'm not sure though.

**Comment by Kuriv2001** [comment]
On demand and availability of invite links, the selection has been changed to: Discord + Telegram + Whatsapp.

File: app/src/main/java/com/monkeyteam/chimpagne/ui/event/ChooseSocialsPanel.kt [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+
+    SocialMediaTextField(
+        url = discordUrl,
+        onUrlChange = { discordUrl = it },
+        labelResource = R.string.discord_group_invite_link,
+        iconResource = R.drawable.discord,
+        testTag = "discord_input",
+        updateSocialMediaLink = { eventViewModel.updateSocialMediaLink(it) },
+        platformName = "discord",
+        platformUrl = "https://discord.gg/")
```
Comment (QuicheLorraine3600) [inline app/src/main/java/com/monkeyteam/chimpagne/ui/event/ChooseSocialsPanel.kt:40]: hardcoded link

File: app/src/main/java/com/monkeyteam/chimpagne/ui/event/ChooseSocialsPanel.kt [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+    Spacer(modifier = Modifier.height(16.dp))
+
+    SocialMediaTextField(
+        url = discordUrl,
+        onUrlChange = { discordUrl = it },
+        labelResource = R.string.discord_group_invite_link,
+        iconResource = R.drawable.discord,
+        testTag = "discord_input",
+        updateSocialMediaLink = { eventViewModel.updateSocialMediaLink(it) },
+        platformName = "discord",
```
Comment (QuicheLorraine3600) [inline app/src/main/java/com/monkeyteam/chimpagne/ui/event/ChooseSocialsPanel.kt:39]: hardcoded name

File: app/src/main/java/com/monkeyteam/chimpagne/ui/event/ChooseSocialsPanel.kt [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+
+    SocialMediaTextField(
+        url = telegramUrl,
+        onUrlChange = { telegramUrl = it },
+        labelResource = R.string.telegram_group_invite_link,
+        iconResource = R.drawable.telegram,
+        testTag = "telegram_input",
+        updateSocialMediaLink = { eventViewModel.updateSocialMediaLink(it) },
+        platformName = "telegram",
+        platformUrl = "https://t.me/")
```
Comment (QuicheLorraine3600) [inline app/src/main/java/com/monkeyteam/chimpagne/ui/event/ChooseSocialsPanel.kt:52]: hardcoded link

File: app/src/main/java/com/monkeyteam/chimpagne/ui/event/ChooseSocialsPanel.kt [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+    Spacer(modifier = Modifier.height(16.dp))
+
+    SocialMediaTextField(
+        url = telegramUrl,
+        onUrlChange = { telegramUrl = it },
+        labelResource = R.string.telegram_group_invite_link,
+        iconResource = R.drawable.telegram,
+        testTag = "telegram_input",
+        updateSocialMediaLink = { eventViewModel.updateSocialMediaLink(it) },
+        platformName = "telegram",
```
Comment (QuicheLorraine3600) [inline app/src/main/java/com/monkeyteam/chimpagne/ui/event/ChooseSocialsPanel.kt:51]: hardcoded name

File: app/src/main/java/com/monkeyteam/chimpagne/ui/event/ChooseSocialsPanel.kt [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+        updateSocialMediaLink = { eventViewModel.updateSocialMediaLink(it) },
+        platformName = "discord",
+        platformUrl = "https://discord.gg/")
+
+    Spacer(modifier = Modifier.height(16.dp))
+
+    SocialMediaTextField(
+        url = telegramUrl,
+        onUrlChange = { telegramUrl = it },
+        labelResource = R.string.telegram_group_invite_link,
```
Comment (QuicheLorraine3600) [inline app/src/main/java/com/monkeyteam/chimpagne/ui/event/ChooseSocialsPanel.kt:47]: hardcoded icon

File: app/src/main/java/com/monkeyteam/chimpagne/ui/event/ChooseSocialsPanel.kt [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+
+    SocialMediaTextField(
+        url = whatsappURL,
+        onUrlChange = { whatsappURL = it },
+        labelResource = R.string.whatsapp_group_invite_link,
+        iconResource = R.drawable.whatsapp,
+        testTag = "whatsapp_input",
+        updateSocialMediaLink = { eventViewModel.updateSocialMediaLink(it) },
+        platformName = "whatsapp",
+        platformUrl = "https://chat.whatsapp.com/")
```
Comment (QuicheLorraine3600) [inline app/src/main/java/com/monkeyteam/chimpagne/ui/event/ChooseSocialsPanel.kt:64]: hardcoded link

File: app/src/main/java/com/monkeyteam/chimpagne/model/database/ChimpagneEvent.kt [hunk 7]
```diff
@@ -23,7 +23,9 @@ data class ChimpagneEvent(
     val ownerId: ChimpagneAccountUID = "",
     val supplies: Map<ChimpagneSupplyId, ChimpagneSupply> = mapOf(),
     val parkingSpaces: Int = 0,
-    val beds: Int = 0
+    val beds: Int = 0,
+    val socialMediaLinks: Map<String, String> =
+        mapOf("discord" to "", "whatsapp" to "", "telegram" to "")
```
Comment (QuicheLorraine3600) [inline app/src/main/java/com/monkeyteam/chimpagne/model/database/ChimpagneEvent.kt:28]: hardcoded values

**Review by QuicheLorraine3600 — CHANGES_REQUESTED** [review]
Nice ! In order to support be able to add social media easily, it would be nice to create a data class representing a `SocialMedia`. Then you could create a `SUPPORTED_SOCIAL_MEDIA: List<SocialMedia>`

**Comment by Kuriv2001** [comment]
Theres is now a SocialMedia data class that can store different social medias with their group links. There is also a supported social media list in there, the only thing that need to be changed now, if a new social media will be supported.

The rest of the code now dynamically takes these values and nothing is hardcoded anymore.

**Comment by Kuriv2001** [comment]
ViewDetailsEventScreen doesn't update directly. Fixed with with Launched Effect

**Review by QuicheLorraine3600 — APPROVED** [review]
Amazing, please solve your conflicts and run tests before merging :)

**Pull request merged**